### PR TITLE
EGL attributes improvements

### PR DESCRIPTION
**Pull request by Daft-Freak**
The main change here is fixing https://github.com/emscripten-ports/SDL2/issues/59 but there are also fixes to querying attributes.

### CODE REVIEW

File: src/library_egl.js [hunk 1]
```diff
@@ -231,6 +234,8 @@ var LibraryEGL = {
       {{{ makeSetValue('value', '0', '0' /* Only pbuffers would be bindable, but these are not supported. */, 'i32') }}};
       return 1;
     case 0x303B: // EGL_MIN_SWAP_INTERVAL
+      {{{ makeSetValue('value', '0', '0', 'i32') }}};
+    return 1;
```
Comment (kripken) [inline src/library_egl.js]: indentation looks wrong here

File: src/library_egl.js [hunk 2]
```diff
@@ -319,25 +324,28 @@ var LibraryEGL = {
       }
       contextAttribs += 8;
     }
+#if USE_WEBGL2
+    if (glesContextVersion < 2 || glesContextVersion > 3) {
```
Comment (kripken) [inline src/library_egl.js]: how does `< 2 || > 3` differ from `!= 2`?

Comment (Daft-Freak) [reply src/library_egl.js]: It allows 3 now.

**Comment by Daft-Freak** [comment]
Looks like I broke legacy emulation.

**Comment by juj** [comment]
Nice! This looks good to me.

Btw, if you are interested in working on EGL side further, there is a work in progress PR at https://github.com/kripken/emscripten/pull/5580 to make EGL support multicontext and multithreading aware, the PR could be used as a basis for that.

**Comment by Daft-Freak** [comment]
Looks like the tests are okay now.

**Pull request merged**